airbyte-config/db-persistence/src/main/java/io.airbyte.config.dbPersistence/PostgresConfigPersistence.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.config.dbPersistence;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.config.ConfigSchema;
+import io.airbyte.db.Databases;
+import io.airbyte.validation.json.JsonSchemaValidator;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.UUID;
+
+public class PostgresConfigPersistence extends DatabaseConfigPersistence {
+
+  public PostgresConfigPersistence(String username, String password, String connectionString) {
+    this(username, password, connectionString, new JsonSchemaValidator());
+  }
+
+  public PostgresConfigPersistence(JsonNode config) {
+    this(config, new JsonSchemaValidator());
+  }
+
+  public PostgresConfigPersistence(JsonNode config, JsonSchemaValidator validator) {
+    this(config.get("username").asText(), config.get("password").asText(), Databases.getPostgresJdbcUrl(config), validator);
```

Comment:
i appreciate the attempt to factor out this configuration component here, but I think this probably a place where we want to keep the code paths separate.

here's how i'm thinking about it... the config (a json blob) that we use in the postgres source, postgres destination, and the database that is used in airbyte-core are all different. at least in the case of source and destination, their configuration objects are declared separately. 

this is to bringing to light one thing, which is putting these naked `JsonNode` in interfaces is kinda dangerous... they're almost no better than just doing type `Object`. this is a bit more tolerable in the connector world because (theoretically) the objects are validated (although as we discussed in your last PR, the tests don't check for it explicitly).

In general where possible we should reduce the use of just `JsonNode`, and I think in this case the schema of the object is knowable enough that we can do that.